### PR TITLE
Add more parameters support in SGEN.

### LINKED ISSUE #22931: Add more parameters, functions in SGEN
Need add the following parameters, functions, which existed in desktop version.

? help
nologo
silent - silently fail
verbose - more error message
debug - keep pdb
keep - keep cs files
parsableerrors
assembly

// For command line apps, cmd.exe cannot display right-to-left languages currently.  Change the
// CurrentUICulture to meet their needs, falling back to our invariant resources.
SetConsoleUICulture
WriteHelp, WriteHeader methods

### LINKED ISSUE #22938: Display warning when fail to generate the serialization code for some types.
Display one warning when fail to generate the serialization code for some types
Put the details in the log

### PR DESCRIPTION
**Pull request by huanwu**
Add following four parameters.

nologo - Prevents displaying of logo.
silent - Prevents displaying of success messages.
verbose - List types from the target assembly that cannot be serialized. This is related to #22938
parsableerrors - Print errors in a format similar to those reported by compilers.

Fix #22931 
Fix #22938

@shmao @zhenlan @mconnew

### CODE REVIEW

File: src/System.Private.Xml/src/Resources/Strings.resx [hunk 1]
```diff
@@ -3399,6 +3399,9 @@
   <data name="InfoNoSerializableTypes" xml:space="preserve">
     <value>Assembly '{0}' does not contain any types that can be serialized using XmlSerializer.</value>
   </data>
+  <data name="InfoIgnoreType" xml:space="preserve">
+    <value>Ignoring '{0}'.</value>
```
Comment (zhenlan) [inline src/System.Private.Xml/src/Resources/Strings.resx]: Is this supposed to be just one type of multiple types? Can you explain what this parameter is for?

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 2]
```diff
@@ -29,6 +29,10 @@ private int Run(string[] args)
             bool force = false;
             bool proxyOnly = false;
             bool disableRun = true;
+            bool nologo = false;
```
Comment (zhenlan) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: In camel casing, this should be `noLogo`. Same for other variables.

File: src/System.Private.Xml/src/Resources/Strings.resx (hunk 1 above)
Comment (huanwu) [reply src/System.Private.Xml/src/Resources/Strings.resx]: This is used in the warning message when we need bypass an unsupported type like generic type.

**Comment by huanwu** [comment]
@dotnet-bot test Tizen armel Debug Build

**Pull request merged**